### PR TITLE
add logs subscriber

### PR DESCRIPTION
**Pull request by schehata**
- Creates a new lambda to automatically create subscription filters on newly created log groups
- The setup utilizes CloudTrail, S3 and EventBridge to to notify the lambda on newly created log groups
- A CloudFormation stack is provided to make it easy to create the setup

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
           mkdir build
           yq ".Resources.LogsLambda.Properties.Code.ZipFile = \"$(sed  's/\"/\\\"/g' ./handler.py)\""  axiom-cloudwatch-lambda-cloudformation-stack.template.yaml > build/axiom-cloudwatch-lambda-cloudformation-stack.yaml
           yq ".Resources.BackfillerLambda.Properties.Code.ZipFile = \"$(sed  's/\"/\\\"/g' ./backfill.py)\""  axiom-cloudwatch-backfiller-lambda-cloudformation-stack.template.yaml > build/axiom-cloudwatch-backfiller-lambda-cloudformation-stack.yaml
+          yq ".Resources.AxiomCloudWatchLogsSubscriber.Properties.Code.ZipFile = \"$(sed  's/\"/\\\"/g' ./backfill.py)\""  axiom-cloudwatch-logs-subscriber-cloudformation-stack.template.yaml > build/axiom-cloudwatch-logs-subscriber-cloudformation-stack.yaml
```
Comment (a-khaledf) [inline .github/workflows/release.yaml:21]: I think you meant ./logs_subscriber.py not ./backfill.py

File: logs_subscriber.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    )
+
+    # Check whether the prefix is set - the prefix is used to determine which logs we want.
+    if not log_group_prefix:
+        create_subscription_filter(
+            log_group_name, log_group_arn, axiom_cloudwatch_lambda_ingester_arn
+        )
+
+        # Check whether the log group's name starts with the set prefix.
+    elif log_group_name.startswith(log_group_prefix):
```
Comment (a-khaledf) [inline logs_subscriber.py:50]: no difference between if log_group_prefix set or not set so you could just put in one condition.
also please add  else -> log that it didn't match the log prefix

**Review by a-khaledf — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (schehata) [reply .github/workflows/release.yaml]: nice catch, fixed.

File: logs_subscriber.py (hunk 2 above)
Comment (schehata) [reply logs_subscriber.py]: done

**Review by a-khaledf — APPROVED** [review]

**Pull request merged**